pennylane/ops/qubit/hamiltonian.py

```diff
@@ -567,7 +567,7 @@ def __mul__(self, a):
         r"""The scalar multiplication operation between a scalar and a Hamiltonian."""
         if isinstance(a, (int, float)):
             self_coeffs = copy(self.coeffs)
-            coeffs = qml.math.multiply(a, self_coeffs)
+            coeffs = qml.math.multiply(qml.math.cast_like([a], self_coeffs), self_coeffs)
```

Comment:
Hi @josh146, just wanted to double-check: are the the changes made to `__mul__` and `__imul__` intentional?

They were part of a bug fix PR (#2376) where e.g., floating scalars were cast to integers due to the use of `qml.math.cast_like`.

Reply:
Oh! I did not change these locally, I wonder why this happened.

Reply:
have removed these!